### PR TITLE
Project suggestions

### LINKED ISSUE #190: Update the StartEditReducer's AutocompleteSuggestionTapped action to handle Project suggestions
## Description
When you select a project suggestion, it sets the project of the editable time entry.
If the project is in a workspace different than the one of the time entry, the time entry workspace is changed as well
Selecting a project suggestion should also remove the word that starts with an @ from the description (do so by checking the cursor position as well)

## Definition of done
- [ ] Tests ensure that the behavior is correct
- [ ] The documentation is updated to reflect the changes done in this PR

### PR DESCRIPTION
**Pull request by heytherewill**
### Summary
This implements a proper search and replace behavior (as per https://github.com/toggl/android/pull/320) for autocomplete suggestions in order to properly handle project suggestions

### Technical considerations
This is the proper use case that covers all corner cases. Feel free to suggest changes to the Swift code, since, well, I'm not the most proficient at it.

### Relationships
Closes #190 

### Pull Request checklist
- [X] The code does not hinder the app's performance
- [X] The code does not introduce regressions
- [X] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [X] The code used custom templates

### Pointers
Review per commit. Check the tests to see what the expected behavior is and if I covered all corner cases. Check if I handled substrings properly as well.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

## Merge permissions
Anyone

### CODE REVIEW

File: Modules/Timer/Timer/StartEdit/Extensions/String+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            let matchedTheFirstWord = tokens.contains { matchSubstring.starts(with: String($0)) }
+            let queryWithToken = String(matchedTheFirstWord ? matchSubstring : matchSubstring.dropFirst())
+            
+            let token = queryWithToken.first
+            let query = String(queryWithToken.dropFirst())
+            
+            return (token, query)
+
+        } catch {
+            return (nil, "")
```
Comment (dshr) [inline Modules/Timer/Timer/StartEdit/Extensions/String+Extensions.swift:37]: Maybe return the original string instead?
```suggestion
            return (nil, self)
```
